data/actions/scripts/other/doors.lua

```diff
@@ -1,3 +1,51 @@
+local positions = {
+	{x = 1, y = 0}, -- east
+	{x = -1, y = 1}, -- south
+	{x = -1, y = -1}, -- west
+	{x = 1, y = -1} -- north
+}
+
+local function moveToAptPos(creature, round)
+	local pos = creature:getPosition()
+	for k,v in pairs(positions) do
+		pos.x = pos.x + v.x
+		pos.y = pos.y + v.y
+		local tile = Tile(pos)
+		if tile then
+			if round <= 2 then
+				if tile:queryAdd(creature, bit.bor(FLAG_IGNOREBLOCKCREATURE, FLAG_PATHFINDING)) == 0 then
+					if tile:getCreatureCount() == 0 and round == 1 then
+						creature:teleportTo(pos, true)
+						return
+					elseif tile:getCreatureCount() > 0 and round == 2 then
+						creature:teleportTo(pos, true)
+						return
+					end
+				end
+			elseif round == 3 then
+				local topItem = tile:getTopDownItem()
+				if topItem then
+					if topItem:getType():isMovable() then
+						creature:teleportTo(pos, true)
+						return
+					end
+				end
+			else
+				if tile:getItemById(ITEM_MAGICWALL) or Tile(pos):getItemById(ITEM_WILDGROWTH) then
+					creature:teleportTo(pos, true)
+					return
+				end
+			end
+		end
+	end
+	if round ~= 4 then
+		return moveToAptPos(creature, round+1)
+	end
+	if creature:isPlayer() then
+		creature:sendTextMessage(MESSAGE_STATUS_SMALL, Game.getReturnMessage(2))
```

Comment:
`creature:sendCancelMessage(RETURNVALUE_NOTENOUGHROOM)`

Reply:
The only way the function would get to that part is if a mapper blocked all 4 exits with immovable items (which I've yet to see any mapper do) so I removed it.